### DISCUSSION TITLE
How to set up component creation without a CommonModule?

### DISCUSSION THREAD

**Discussion by iSpid3r** [cyrilletuzi/vscode-angular-schematics#930] [Q&A] [answered]
Hello.

Why is there no such option?

**Comment by cyrilletuzi** [comment]
Hello,

From your [issue](https://github.com/cyrilletuzi/vscode-angular-schematics/issues/932):
> Although in the new CLI they are created without it

The official Angular CLI can manage this more easily because its version is tied to the Angular version of the project. It is more complex for the extension to manage such changing Angular features, because the extension is not 100% sure of the Angular version of the project (due to the nightmare of all the existing package managers fantasies).

Also, some people may want to stay on legacy features in legacy projects.

> or at least have the ability to customize it

For information, the option already exists in the Pro edition. Managing all the cases (new features, Angular versions, people wanting to stay on legacy features on legacy projects despite being on a newer version, etc.) has become a nightmare too, so such advanced developments are reserved to the Pro edition now.

So for all these reasons, I was waiting for the new control flow (ie. no more CommonModule) to be officially stable in LTS versions, so it is safe to make it the default in the standard edition. It will be the case soon, as Angular 16 will end its LTS in November, when Angular 19 arrives. So I will look and change that soon, in addition to standalone components becoming the default in Angular 19.

**Comment by cyrilletuzi** [comment, marked as answer]
Hello, for information, the default `CommonModule` import has been removed in version 6.18.
